packages/liveblocks-node/src/client.ts

```diff
@@ -1246,6 +1248,64 @@ export class Liveblocks {
     }
   }
 
+  /**
+   * Mark a thread as resolved.
+   * @param params.roomId The room ID to update the thread in.
+   * @param params.threadId The thread ID to update.
+   * @param params.data.userId The user ID of the user who updated the thread.
+   * @returns The updated thread.
+   */
+  public async markThreadAsResolved(params: {
+    roomId: string;
+    threadId: string;
+    data: {
+      userId: string;
+    };
+  }): Promise<M> {
+    const { roomId, threadId } = params;
+
+    const res = await this.post(
+      url`/v2/rooms/${roomId}/threads/${threadId}/mark-as-resolved`,
+      {}
+    );
+
+    if (!res.ok) {
+      const text = await res.text();
+      throw new LiveblocksError(res.status, text);
+    }
+
+    return (await res.json()) as M;
```

Comment:
Oh I missed this, the endpoints don't return the thread's metadata no?

Reply:
Oh yeah, that's indeed a bug, good catch 👍

Reply:
Let's add type-level tests for this while we're fixing this, to avoid regressions. Here are the relevant tests for `editThreadMetadata`:

- When [using augmentation](https://github.com/liveblocks/liveblocks/blob/main/packages/liveblocks-node/test-d/augmentation.test-d.ts#L256-L312)
- When [not using augmentation](https://github.com/liveblocks/liveblocks/blob/main/packages/liveblocks-node/test-d/no-augmentation.test-d.ts#L207-L245)

@ofoucherot For context, in case you have never updated these type-level tests: the files inside `test-d/*` are static tests only. The code you write in there isn't actually (going to be) executed. However, you can add `expectType<Foo>(someExpression)` assertions in there, to ensure that calling `someExpression` will return a value of a specific type (in this case `Foo`). To run these, run `turbo run test:types` in any package.

Reply:
Edit thread metadata only returns the metadata. I'm fixing the two new endpoints' signature.